### PR TITLE
Session scoped site matrix

### PR DESCRIPTION
**Pull request by phlax**
~~**ive made this WIP again while i do some further testing/refactoring**~~

### CODE REVIEW

File: pytest_pootle/env.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    methods = (
+        "case_sensitive_schema", "content_type", "site_root", "system_users",
+        "permissions", "site_permissions", "tps", "languages", "vfolders",
+        "subdirs", "announcements")
+
+    def __init__(self, request):
+        self.request = request
+
+    @cached_property
+    def dirs(self):
```
Comment (dwaynebailey) [inline pytest_pootle/env.py]: Codacy Issue found: [An attribute defined in pytest_pootle.env line 29 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1959294923/issues/source?bid=1990595&fileBranchId=3109045#l26)

Comment (dwaynebailey) [inline pytest_pootle/env.py]: Codacy Issue found: [An attribute defined in pytest_pootle.env line 29 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1959294923/issues/source?bid=1990595&fileBranchId=3109045#l26)

File: pytest_pootle/env.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
```
Comment (dwaynebailey) [inline pytest_pootle/env.py]: Codacy Issue found: [An attribute defined in pytest_pootle.env line 29 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1968005668/issues/source?bid=1990595&fileBranchId=3109045#l26)

Comment (dwaynebailey) [inline pytest_pootle/env.py]: Codacy Issue found: [An attribute defined in pytest_pootle.env line 29 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1969460327/issues/source?bid=1990595&fileBranchId=3109045#l26)

Comment (dwaynebailey) [inline pytest_pootle/env.py]: Codacy Issue found: [An attribute defined in pytest_pootle.env line 29 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1971021410/issues/source?bid=1990595&fileBranchId=3109045#l26)

Comment (dwaynebailey) [inline pytest_pootle/env.py]: Codacy Issue found: [An attribute defined in pytest_pootle.env line 29 hides this method](https://www.codacy.com/app/dwayne/pootle/file/1972911443/issues/source?bid=1990595&fileBranchId=3109045#l26)

**Comment by unho** [comment]
lgtm

**Pull request merged**